src/app/todo/page.tsx

```diff
@@ -0,0 +1,64 @@
+"use client";
+
+import { useRedirect } from "@/hooks/useRedirect";
+import { useUser } from "@/hooks/useUser";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+
+export default function Todo() {
+  const [newTodo, setNewTodo] = useState("");
+  const [todos, setTodos] = useState<any[]>([]);
+  const [metaData, setMetadata] = useState({ owner: "J.Doe", count: 0 });
+
+  const user = useUser();
+
+  if (!user.loggedIn) {
+    useRedirect();
+  }
+
+  const addTodo = () => {
+    setTodos([...todos, { key: uuid(), name: newTodo, completed: false }]);
+    setNewTodo("");
+    metaData.count++;
+  };
+
+  const completeTodo = (key: string) => {
+    const newTodos = [...todos];
+    console.log(key);
+    const todo = newTodos.find((t) => (t.key = key));
+    console.log(todo);
+    todo.completed = true;
+    setTodos(newTodos);
```

Comment:
The completeTodo function should not be finding the todo item by its key. Instead, it should be using the findIndex function to find the index of the todo item in the todos array.